### PR TITLE
Puli

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
     ],
     "require": {
         "php": ">=5.4",
-        "php-http/httplug": "1.0.0-alpha3",
-        "php-http/message-factory": "^1.0"
+        "php-http/httplug": "dev-puli",
+        "php-http/message-factory": "dev-puli"
```
Comment (dbu) [inline composer.json]: wrap up before merge

File: puli.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (dbu) [inline puli.json:1]: is this file supposed to be committed? what about other factories, can they come with their own puli file? and should the binding not be in either discovery or the adapter repo?

Comment (dbu) [inline puli.json:1]: note that i am not familiar with puli, so i just ask the stupid questions :-)

Comment (sagikazarmark) [inline puli.json:1]: This is the binding. The binding types which declare these bindings are in httplug and message-factory. Each package come with its own puli.json file. It contains data about bindings and resources and is read by puli when you install it in an application. See the puli-test repo for usage.

**Pull request merged**